### PR TITLE
adds in a 'no eorg' preference that teleports you to the ghost cafe in round end

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

see title

## Why It's Good For The Game

Because people get salty over eorg and this is one way to stop it, at least for a while until I have to code in some edge case to stop them from eorging in the ghost cafe.

## Changelog
:cl:
add: Ability to be transported to the ghost cafe on roundend instead of eorg
/:cl:

### CODE REVIEW

File: code/__HELPERS/roundend.dm [hunk 1]
```diff
@@ -191,7 +191,20 @@
 			H.add_hud_to(M)
 
 	CHECK_TICK
-
+	for(var/mob/m in GLOB.mob_list)
```
Comment (Ghommie) [inline code/__HELPERS/roundend.dm]: One, all atoms in mob_list should be mobs, so the typecheck is not necessary versus a new line `var/mob/M = m`.
Two, it'll tele people left stranded on the station to the ghost cafe too.
Third, I think it should be better if you used the player_list.

File: code/__HELPERS/roundend.dm [hunk 2]
```diff
@@ -191,7 +191,20 @@
...
+		if (m.client)
```
Comment (Ghommie) [inline code/__HELPERS/roundend.dm]: Early return if they don't have a client.

File: code/__HELPERS/roundend.dm [hunk 3]
```diff
@@ -191,7 +191,20 @@
...
+			m.client.verbs += /client/proc/tele_to_ghost_cafe
+		if (m.client && m.client.prefs && m.client.prefs.eorg_tele)
+			ADD_TRAIT(m,TRAIT_PACIFISM,ROUNDSTART_TRAIT)
+			if (ishuman(m))
+				var/mob/living/carbon/human/H = m
+				H.revive(full_heal=TRUE,admin_revive=TRUE)
+				m.status_flags |= GODMODE
+				for (var/obj/item/W in H)
```
Comment (Ghommie) [inline code/__HELPERS/roundend.dm]:
```
				for (var/i in H.get_equipped_items(TRUE))
					var/obj/item/I = i
```

File: code/__HELPERS/roundend.dm [hunk 4]
```diff
@@ -191,7 +191,20 @@
 			H.add_hud_to(M)
 
 	CHECK_TICK
-
+	for(var/mob/m in GLOB.mob_list)
+		if (m.client)
+			m.client.verbs += /client/proc/tele_to_ghost_cafe
+		if (m.client && m.client.prefs && m.client.prefs.eorg_tele)
+			ADD_TRAIT(m,TRAIT_PACIFISM,ROUNDSTART_TRAIT)
+			if (ishuman(m))
```
Comment (Ghommie) [inline code/__HELPERS/roundend.dm]: the human check is only necessary for the uniform and shoes. bring it down there.

File: code/__HELPERS/roundend.dm [hunk 5]
```diff
@@ -191,7 +191,20 @@
...
+				var/mob/living/carbon/human/H = m
+				H.revive(full_heal=TRUE,admin_revive=TRUE)
+				m.status_flags |= GODMODE
+				for (var/obj/item/W in H)
+					if (W == H.w_uniform || W == H.shoes)
+						continue
+					H.dropItemToGround(W)
```
Comment (Ghommie) [inline code/__HELPERS/roundend.dm]: Oh yea, btw this will light plasmamemes on fire.

**Review by Ghommie — CHANGES_REQUESTED** [review]
I have a certain perplexity on this feature.
Under normal conditions, the game will restart in 1 minute anyway, so roleplay is irrelevant.
On delayed restarts, people may assume the cafe as a no EORG zone and ahelp the grievers while the admins are busy dealing with other tickets.

File: code/__HELPERS/roundend.dm (hunk 1 above)
Comment (Putnam3145) [reply code/__HELPERS/roundend.dm]: For reference: the second line is faster than the typecheck. You should prefer it when it's possible to do.

File: code/__HELPERS/roundend.dm (hunk 4 above)
Comment (Putnam3145) [reply code/__HELPERS/roundend.dm]: Also, this shouldn't be ROUNDSTART_TRAIT added, though it's kind of hilariously irrelevant

File: code/__HELPERS/roundend.dm (hunk 3 above)
Comment (Seris02) [reply code/__HELPERS/roundend.dm]: why though

File: code/__HELPERS/roundend.dm (hunk 4 above)
Comment (Seris02) [reply code/__HELPERS/roundend.dm]: and the revival but ok

File: code/__HELPERS/roundend.dm (hunk 5 above)
Comment (Seris02) [reply code/__HELPERS/roundend.dm]: true.

File: code/__HELPERS/roundend.dm (hunk 3 above)
Comment (Ghommie) [reply code/__HELPERS/roundend.dm]: cavity implants, and any other item that's not equipped but found inside the mob anyway.

Comment (Seris02) [reply code/__HELPERS/roundend.dm]: it works for the ninja net though but alright.

**Comment by silicons** [comment]
this is just hugbox

**Comment by Seris02** [comment]
basically yeah

File: code/__HELPERS/roundend.dm (hunk 1 above)
Comment (Seris02) [reply code/__HELPERS/roundend.dm]: I want it to tele people on the station with this on, and it does it after it collects the roundend report, so it shouldn't affect anything.

**Comment by Ghommie** [comment]
This is irrelevant, you can still use the ghost cafe in case of delayed round ends.

**Pull request closed without merging**